lib/deep-equal.js

```diff
@@ -51,11 +51,11 @@ function deepEqualCyclic(first, second, match) {
         // If both are matchers they must be the same instance in order to be
         // considered equal If we didn't do that we would end up running one
         // matcher against the other
-        if (match && match.isMatcher(obj1)) {
-            if (match.isMatcher(obj2)) {
+        if (match && match.isMatcher(obj2)) {
+            if (match.isMatcher(obj1)) {
                 return obj1 === obj2;
             }
-            return obj1.test(obj2);
+            return obj2.test(obj1);
```

Comment:
Were any changes in this file necessary? It seems like you just switched variable names, but I'm not seeing the why.

Reply:
Good question. It's easier to see if you review each commit individually. I had to introduce this check on top of the previous one to make these new test cases pass:

https://github.com/sinonjs/samsam/blob/2760e751575ae0f318ceaa790fca3d19cc67dda8/lib/sinon-match-integration.test.js#L50-L62

It then occurred to me, that we can have the same simple solution as before with the tests still passing if we swapped the variable names.